pkgs/build-support/ocaml/dune.nix

```diff
@@ -3,7 +3,15 @@
 { pname, version, nativeBuildInputs ? [], enableParallelBuilding ? true, ... }@args:
 
 let Dune =
-  let dune-version = args . duneVersion or (if args.useDune2 or true then "2" else "1"); in
+  let dune-version =
+    args.duneVersion or (
+        if builtins.hasAttr "useDune2" args then
+          if args.useDune2 or true then
+            builtins.trace "`useDune2 = true` is deprecated please consider replacing it with duneVersion = \"2\"" "2"
+          else
+            builtins.trace "`useDune2 = false` is deprecated please consider replacing it with duneVersion = \"1\"" "1"
+        else "2"
+      ); in
```

Comment:
I tested this change with this file:

dune.nix:

```nix
dune = args: args.duneVersion or (if args.useDune2 or true then "2" else "1")
dune = args:
    args.duneVersion or (
        if builtins.hasAttr "useDune2" args then
          if args.useDune2 or true then
            builtins.trace "`useDune2 = true` is deprecated please consider replacing it with duneVersion = \"2\"" "2"
          else
            builtins.trace "`useDune2 = false` is deprecated please consider replacing it with duneVersion = \"1\"" "1"
        else "2"
      )

assert (dune { }) == "2"; true
assert (dune { useDune2 = true; }) == "2"; true 
assert (dune { useDune2 = false; }) == "1"; true

assert (dune { duneVersion = "1"; }) == "1"; true
assert (dune { duneVersion = "1"; useDune2 = true; }) == "1"; true
assert (dune { duneVersion = "1"; useDune2 = false; }) == "1"; true

assert (dune { duneVersion = "2"; }) == "2"; true
assert (dune { duneVersion = "2"; useDune2 = true; }) == "2"; true
assert (dune { duneVersion = "2"; useDune2 = false; }) == "2"; true

assert (dune { duneVersion = "3"; }) == "3"; true
assert (dune { duneVersion = "3"; useDune2 = true; }) == "3"; true
assert (dune { duneVersion = "3"; useDune2 = false; }) == "3"; true
```

and this command `nix repl < dune.nix 2>&1 | grep -v TCSADRAIN` I see no assert triggered ans see only two trace.

If you uncomment the second implementation you get the same result. I enumerated all possible case (duneVersion: absent, "1", "2" and "3", useDune2: absent, true, false). So I am confident this change will doesn't cause regression.